optics-core/src/Optics/Internal/Optic/Subtyping.hs

```diff
@@ -26,17 +27,71 @@ class Is k l where
   -- | Witness of the subtyping relationship.
   implies :: (Constraints k p => r) -> (Constraints l p => r)
 
--- | Overlappable instance for a custom type error.
-instance {-# OVERLAPPABLE #-} TypeError ('ShowType k
-                                         ':<>: 'Text " cannot be used as "
-                                         ':<>: 'ShowType l
-                                        ) => Is k l where
-  implies _ = error "unreachable"
-
 -- | Every kind of optic can be used as itself.
 instance Is k k where
   implies r = r
 
+-- | Overlappable instance for a custom type error.
+instance {-# OVERLAPPABLE #-} TypeError
+  ('ShowType k ':<>: 'Text " cannot be used as " ':<>: 'ShowType l
+   ':$$: 'Text "Perhaps you meant to use one of these:"
+   ':$$: EliminationForms k
+  ) => Is k l where
+  implies _ = error "unreachable"
+
+type family EliminationForms (k :: OpticKind) :: ErrorMessage where
+  EliminationForms An_AffineFold      = AffineFoldEliminations
+  EliminationForms An_AffineTraversal = AffineTraversalEliminations
+  EliminationForms A_Fold             = FoldEliminations
+  EliminationForms A_Getter           = GetterEliminations
+  EliminationForms A_Lens             = LensEliminations
+  EliminationForms A_Prism            = PrismEliminations
+  EliminationForms A_ReversedLens     = PrismEliminations
+  EliminationForms A_ReversedPrism    = LensEliminations
+  EliminationForms A_Review           = ReviewEliminations
+  EliminationForms A_Setter           = SetterEliminations
+  EliminationForms A_Traversal        = TraversalEliminations
```

Comment:
This isn't total, it is missing `An_Iso`. I guess one doesn't need it as `Iso` can be used as anything, but I'd like `type family` to be total anyway.